rest/data/perf.go

```diff
@@ -233,6 +236,65 @@ func (dbc *DBConnector) FindPerformanceResultWithChildren(ctx context.Context, i
 	return apiResults, nil
 }
 
+// ScheduleSignalProcessingRecalculateJobs schedules signal processing recalculation jobs for
+// each project/version/task/test combination
+func (dbc *DBConnector) ScheduleSignalProcessingRecalculateJobs(ctx context.Context) error {
+	db := dbc.env.GetDB()
+	pipeline := bson.A{
+		bson.M{
+			"$match": bson.M{
+				"info.order": bson.M{
+					"$exists": true,
+				},
+				"info.mainline":  true,
+				"info.project":   bson.M{"$exists": true},
+				"info.variant":   bson.M{"$exists": true},
+				"info.task_name": bson.M{"$exists": true},
+				"info.test_name": bson.M{"$exists": true},
+			},
+		},
+		bson.M{
+			"$group": bson.M{
+				"_id": bson.M{
+					"task":    "$info.task_name",
+					"variant": "$info.variant",
+					"project": "$info.project",
+					"test":    "$info.test_name",
+				},
+			},
+		},
+	}
+	cur, err := db.Collection("perf_results").Aggregate(ctx, pipeline)
+	defer cur.Close(ctx)
+	if err != nil {
+		return gimlet.ErrorResponse{StatusCode: http.StatusInternalServerError, Message: fmt.Sprint("Failed to aggregate recalculation metrics")}
+	}
+	for cur.Next(ctx) {
+		var result units.MetricGrouping
```

Comment:
put the declaration outside of the loop? maybe make it a pointer? in the declaration?

Reply:
moved it out